src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -11,44 +11,39 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             uint flags = Flags;
             CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
-            const int StackAllocThreshold = 512;
-            if (count < StackAllocThreshold)
-            {
-                char* output = stackalloc char[count];
-                return GetAsciiCore(unicode, count, flags, output, count, reattempt: true);
-            }
-            else
+            const int StackallocThreshold = 512;
+            // Each unicode character is represented by 2 ASCII chars
```

Comment:
Don't worry, its important to get right - I updated this to "up to 3 ASCII chars", which I think is correct taking into account some of the data that previously didn't match this.
I'm happy to fix again if I'm wrong